### DISCUSSION TITLE
[c64] restarting program from NMI reset (or exit() ) does not restart program.

### DISCUSSION THREAD

**Discussion by tschak909** [cc65/cc65#2103] [Q&A]
What is preventing a cc65 program from restarting from an NMI or restart after exit() ?

**Comment by mrdudz** [comment]
Without more info that is impossible to answer - however, on some platforms (like the c64) the .data section (initialized variables that can be modified at runtime) is used in-place after the binary was loaded. That means when the program modifies those, they will be different on another run. This makes it impossible to re-run those programs on those platforms, you must load them again. Or write your program in a way that it will take care of this :)
